src/components/common/button-icon/button-icon.tsx

```diff
@@ -1,49 +1,56 @@
-import React, { useCallback, VFC } from 'react';
+import React, { FC, useMemo } from 'react';
 import { MaterialCommunityIcons } from '@expo/vector-icons';
-import { TouchableOpacity } from 'react-native';
+import { StyleProp, TextStyle, TouchableOpacity, TouchableWithoutFeedback } from 'react-native';
 
 import globalStyles from '~global/constants.style';
-import { TodoButtonsNameType } from '~types/todo.types';
+import { IconsNames } from '~types/todo.types';
+import styles from './button-icon.style';
+import { iconNamePicker } from './utils/iconNamePicker';
 
-export const ButtonIcon: VFC<Props> = ({ onPress, variant }) => {
-  const iconNamePicker = useCallback(
-    (
-      btnVariant: typeof variant,
-    ): {
-      name: TodoButtonsNameType;
-      color: string;
-    } => {
-      switch (btnVariant) {
-        case 'add':
-          return { name: 'plus-circle', color: globalStyles.SUCCESS_COLOR };
-        case 'delete':
-          return { name: 'delete-circle', color: globalStyles.DELETE_COLOR };
-        case 'save':
-          return { name: 'content-save', color: globalStyles.LIGHT_MAIN_COLOR };
-        case 'select-all':
-          return {
-            name: 'checkbox-multiple-marked-circle',
-            color: globalStyles.SUCCESS_COLOR,
-          };
-        default:
-          return { name: 'cancel', color: globalStyles.LIGHT_CANCEL_COLOR };
-      }
-    },
-    [],
-  );
-
-  return (
-    <TouchableOpacity onPress={onPress} accessibilityRole="button">
+export const ButtonIcon: FC<Props> = ({
+  onPress,
+  variant,
+  size,
+  style,
+  testID,
+  hasOpacity = true,
+}) => {
+  const Icon = useMemo(() => {
+    return (
       <MaterialCommunityIcons
         name={iconNamePicker(variant).name}
-        size={globalStyles.ICON_MED_SIZE}
         color={iconNamePicker(variant).color}
+        size={size ?? globalStyles.ICON_MED_SIZE}
+        style={style}
+        testID={testID}
       />
-    </TouchableOpacity>
+    );
+  }, [variant, size, testID, style]);
+
+  return (
+    <>
+      {hasOpacity ? (
+        <TouchableOpacity onPress={onPress} accessibilityRole="button" style={styles.wrapper}>
+          {Icon}
+        </TouchableOpacity>
+      ) : (
+        <TouchableWithoutFeedback
+          onPress={onPress}
+          accessibilityRole="button"
+          style={styles.wrapper}
```

Comment:
Same props in both components can be moved to one object and then put into both of them
{...componentsProps}

Reply:
Moved to one object named 'componentsProps'